lib/charms/lego_base_k8s/v0/lego_client.py

```diff
@@ -316,6 +320,22 @@ def _cmd(self) -> List[str]:
             "run",
         ]
 
+
+
+    @property
+    def _app_environment(self) -> Dict[str, str]:
+        """Extract proxy model environment variables."""
+        env = {}
+
+        if (http_proxy := get_env_var(env_var="HTTP_PROXY")):
+            env["HTTP_PROXY"] = http_proxy
+        if (https_proxy := get_env_var(env_var="HTTPS_PROXY")):
+            env["HTTPS_PROXY"] = https_proxy
+        # there's no need for no_proxy if there's no http_proxy or https_proxy
```

Comment:
I added the extra condition `and (http_proxy and https_proxy):` following the example in github-runner-operator [here](https://github.com/canonical/github-runner-operator/blob/82067e501bbb9e7e23330c433098b4083beaa433/src/charm_state.py#L688).